rep-2011.rst

```diff
@@ -877,6 +877,22 @@ Similar to the integration with launch, you could run ``ros2 bag play ...`` and
 
 That option would handle the conversion on the fly using the same mechanisms as the command line tool.
 
+Security Concerns
+=================
+
+The additional features this REP proposes do not impose significantly increased security concerns, since the message conversions happen over topics and services.
+
+More precisely, transfer functions would just be another node that interfaces with topics and services to do the message conversions, and hence would be secured by the same security mechanisms that secure nodes, topics, and services in general.
+Furthermore, transfer functions would have to be defined ahead of time (local to the machine running the transfer function), and cycles in the transfer function chain would throw assertions, making it less feasible to use the transfer function feature as an attack vector.
+The transfer functions would also generally need to be opted into by the user (e.g. in a launch file or otherwise explicitly invoked), further reducing the attack surface.
+
+Likewise, for the version hashing component, the hash gets appended to the type name that is sent via discovery.
+This means that the same security mechanisms that secure topic type names sent on discovery will secure the version hash, so there is no increase in attack surface.
+
+Of course, it could be argued that since the version hash is a hashed version of some parsed version of the type description, there is an increase in the kinds of information that an attacker would gain access to if the hash was crackable/reversible, since information about the message's structure would then be obtained.
```

Comment:
This then contradicts the previous statement. So i'd suggest merging both and reasoning about why this does not imply a high risk (which is essentially argued below).